src/sequence.js

```diff
@@ -1046,7 +1093,97 @@ const zipLongest = curry('zipLongest', (seqs, fallback) => pipe(
   map(list),
 ));
 
+/** Curryable version of zipLongest */
+const zipLongest2 = curry('zipLongest2', (a, b, fallback) => zipLongest([a, b], fallback));
+
+/**
+ * Forms a sliding window on the underlying iterator.
+ *
+ * `slidingWindow([1,2,3,4,5], 3)`
+ * yields `[[1,2,3], [2,3,4], [3,4,5]]`
+ *
+ * Will throw IteratorEnded if the sequence is shorter than
+ * the given window.
+ *
+ * @param {Sequence} seq A sequence of sequences
+ * @returns {Iterator} Iterator of lists
+ */
+const slidingWindow = curry('slidingWindow', (seq, no) => {
+  const it = iter(seq);
+  const cache = [];
+  each(range0(no), () => cache.push(next(it)));
+
+  // By making just this part a generator we make
+  // sure that an error while filling the cache
+  // above is thrown early
+  function* slidingWindowImpl() {
+    yield list(cache);
+    for (const v of it) {
+      cache.shift();
+      cache.push(v);
+      yield list(cache);
+    }
+  }
+
+  return slidingWindowImpl();
+});
+
+/**
+ * Like slidingWindow, but returns an empty sequence if the given
+ * sequence is too short.
+ */
+const trySlidingWindow = curry('trySlidingWindow', (seq, no) => {
+  try {
+    return slidingWindow(seq, no);
+  } catch (e) {
+    if (e instanceof IteratorEnded) {
+      return iter([]);
+    }
+    throw e;
```

Comment:
Is this even possible? Can you provide a test case or should this always return an empty sequence if there's an error accessing the sequence items?

Reply:
You mean the throw e?
In this specific example there is quite a few cases where, an error might be raised:

The underlying slidingWindow implementation uses next() and iter() during setup. next() and iter() basically translate to arbitrary function calls, so in these anything can happen, including terminating the program and throwing arbitrary exceptions.
iter() will also throw errors for null/undefined and for any values not implementing the Sequence protocol.

There also is the `[]` and `.push()`; I haven't found anything specific on the exception guarantees of the array constructor & push, so I am generally assuming they might. E.g. [] might throw if the system allocator fails to allocate a memory block  and I have been able to verify that the array constructor and push throw if the array length grows beyond 2**32-1 items, so they are not forbidden from throwing.
Admittedly, both is unlikely but I haven't been able to ascertain that they don't throw other errors…

Then js also handles most syntax errors using exceptions; e.g. if at any place a const variable was redeclared, that would also result in a SyntaxError being thrown and for debuggin purposes I wouldn't want to hide these errors either…

So given this level of exception unsafety, I use type checks when catching errors as a best practice unless I really want a catch all…

Although actually, this code above is not even fully correct, since the underlying iterator may also throw IteratorEnded, so I'll replace this with a version that does not rely on catching exceptions…